pkg/sysfs/system.go

```diff
@@ -1183,6 +1183,15 @@ func (sys *system) discoverNodes() error {
 		return fmt.Errorf("failed to parse nodes with memory (%q): %v",
 			memoryNodeIDs, err)
 	}
+	onlineNodeIDs, err := readSysfsEntry(sysNodesPath, "online", nil)
```

Comment:
There seems to be a corner case: if there is a memoryless NUMA node but its CPUs are offlined, Linux still shows the node in the "online" list (tested with linux 6.5.6-300.fc39.x86_64). As a result, this logic includes the node in the noMemNodes list, as through that into dramNodes list, too.

On line 1212 we calculate dramNodes as "memory nodes with CPUs + nodes without memory". I'd suggest to simplify this as "nodes with CPUs", and fill it from the "has_cpu" list. As a result we would avoid getting nodes without (online) CPUs and memory included in dramNodes and thereby avoid accidentally pinning workloads without cpu/memory requests into a node that has neither.

Reply:
Since you want to filter out noMemNodes without CPUs, I think it would be much more straightforward to just change line 1212 from the 
current intersection+union `dramNodes := memoryNodes.Intersection(cpuNodes).Union(noMemNodes)`
to first union then intersect `dramNodes := memoryNodes.Union(noMemNodes).Intersection(cpuNodes)`

WDYT ?

Reply:
Isn't this the same as `dramNodes := cpuNodes`?

Reply:
Hmm... yes, I think it should be.